packages/libs/eda/src/lib/map/analysis/hooks/standaloneMapMarkers.tsx

```diff
@@ -0,0 +1,598 @@
+        const position = { lat: avgLat, lng: avgLon };
+
+        const donutData =
+          vocabulary && overlayValues && overlayValues.length
+            ? overlayValues.map(({ binLabel, value }) => ({
+                label: binLabel,
+                value: value,
+                color:
+                  overlayType === 'categorical'
+                    ? ColorPaletteDefault[vocabulary.indexOf(binLabel)]
+                    : gradientSequentialColorscaleMap(
+                        vocabulary.indexOf(binLabel) / (vocabulary.length - 1)
+                      ),
+              }))
+            : [];
+
+        // TO DO: address diverging colorscale (especially if there are use-cases)
+
+        // now reorder the data, adding zeroes if necessary.
+        const reorderedData =
+          vocabulary != null
+            ? vocabulary.map(
+                (
+                  overlayLabel // overlay label can be 'female' or a bin label '(0,100]'
+                ) =>
+                  donutData.find(({ label }) => label === overlayLabel) ?? {
+                    label: fixLabelForOtherValues(overlayLabel),
+                    value: 0,
+                  }
+              )
+            : // however, if there is no overlay data
+              // provide a simple entity count marker in the palette's first colour
+              [
+                {
+                  label: 'unknown',
+                  value: entityCount,
+                  color: '#333',
+                },
+              ];
+
+        const MarkerComponent =
+          markerType == null || markerType === 'pie'
+            ? DonutMarker
+            : ChartMarker;
+
+        // TO DO: this will sum to 1 when we allow proportion mode
+        // maybe ask for the counts to come along site the values in the back end response
+        // https://github.com/VEuPathDB/EdaDataService/issues/261
+        // (the way we did it in the old viz wasn't 100% accurate)
+        const count = reorderedData.reduce(
+          (sum, item) => (sum = sum + item.value),
+          0
+        );
+
+        const formattedCount =
+          MarkerComponent === ChartMarker
+            ? mFormatter(count)
+            : kFormatter(count);
+
+        return (
+          <MarkerComponent
+            id={geoAggregateValue}
+            key={geoAggregateValue}
+            bounds={bounds}
+            position={position}
+            data={reorderedData}
+            duration={defaultAnimationDuration}
+            markerLabel={miniMarkers ? '' : formattedCount}
+            {...(markerType !== 'pie'
+              ? {
+                  dependentAxisRange: defaultDependentAxisRange,
+                  dependentAxisLogScale: dependentAxisLogScale,
+                }
+              : {})}
+            {...(miniMarkers && !invisibleMarkers
+              ? { markerScale: 0.5 }
+              : invisibleMarkers
+              ? { markerScale: 0 }
+              : {})}
+          />
+        );
```

Comment:
Will do. Talking of scaling, we now have two different hooks for markers - this one and the old `useMapMarkers` (for the EDA viz)

If we add new markers to the SAM it will be extra work to add them to the EDA viz.

Maybe we port the EDA to use the new hook - that's probably easiest.

Reply:
BTW, I'm fine with addressing the original comment as a follow-up, so that we can get this PR merged sooner. Up to you.

Reply:
I've addressed the original comment (refactor to switch statement) - so can we approve and merge?

Reply:
Yes this data hook doesn't return birds eye plot data :-(